parcels/particleset.py

```diff
@@ -163,11 +167,16 @@ def ArrayClass_init(self, *args, **kwargs):
             raise NotImplementedError("If fieldset.time_origin is not a date, time of a particle must be a double")
         time = np.array([self.time_origin.reltime(t) if _convert_to_reltime(t) else t for t in time])
         assert lon.size == time.size, "time and positions (lon, lat, depth) do not have the same lengths."
-        if isinstance(fieldset.U, Field) and (not fieldset.U.allow_time_extrapolation):
-            _warn_particle_times_outside_fieldset_time_bounds(time, fieldset.U.grid.time_full)
 
-        if lonlatdepth_dtype is None:
-            lonlatdepth_dtype = self.lonlatdepth_dtype_from_field_interp_method(fieldset.U)
+        if type(fieldset) == UXFieldSet:
+            lonlatdepth_dtype = np.float32  # To do : get precision from fieldset
+        else:
+            if isinstance(fieldset.U, Field) and (not fieldset.U.allow_time_extrapolation):
+                _warn_particle_times_outside_fieldset_time_bounds(time, fieldset.U.grid.time_full)
```

Comment:
I suppose we'll need an updated implementation of `_warn_particle_times_outside_fieldset_time_bounds` .. Alternatively, we'll need to replace the `fieldset.U.grid.time_full` with a `fieldset.time_full` call.

Note that I'm assuming that a UXarray or XArray dataset have a single time dimension for all fields in the dataset. Getting the time extents for the dataset wouldn't depend on the individual fields (e.g. `U` or `V`); rather it'd be associated with the dimensions of the dataset.